source/grid/grid_generator.cc

```diff
@@ -1910,6 +1910,353 @@ namespace GridGenerator
+                           const double             pad_right,
+                           const Point<2>           new_center,
+                           const types::manifold_id polar_manifold_id,
+                           const types::manifold_id tfi_manifold_id,
+                           const double             L,
+                           const unsigned int /*n_slices*/,
+                           const bool colorize)
+    {
+      Assert((pad_bottom > 0 || pad_top > 0 || pad_left > 0 || pad_right > 0),
+             ExcMessage("At least one padding parameter has to be non-zero."));
+      Assert(pad_bottom >= 0., ExcMessage("Negative bottom padding."));
+      Assert(pad_top >= 0., ExcMessage("Negative top padding."));
+      Assert(pad_left >= 0., ExcMessage("Negative left padding."));
+      Assert(pad_right >= 0., ExcMessage("Negative right padding."));
+
+      const Point<2> center;
+
+      auto min_line_length = [](const Triangulation<2> &tria) -> double {
+        double length = std::numeric_limits<double>::max();
+        for (const auto cell : tria.active_cell_iterators())
+          for (unsigned int n = 0; n < GeometryInfo<2>::lines_per_cell; ++n)
+            length = std::min(length, cell->line(n)->diameter());
+        return length;
+      };
+
+      // start by setting up the cylinder triangulation
+      Triangulation<2> cylinder_tria;
+      GridGenerator::hyper_cube_with_cylindrical_hole(cylinder_tria,
+                                                      inner_radius,
+                                                      outer_radius,
+                                                      L,
+                                                      /*repetitions*/ 1,
+                                                      colorize);
+
+      // Mark the cylinder material ids as 2 to distinguish them from the bulk
+      // cells (these are read again about 40 lines down)
+      for (const auto &cell : cylinder_tria.active_cell_iterators())
+        cell->set_material_id(2);
+
+      // hyper_cube_with_cylindrical_hole will have 2 cells along
+      // each face, so he element size is outer_radius
+
+      auto add_sizes = [](std::vector<double> &step_sizes,
+                          const double         padding,
+                          const double         h) -> void {
+        // use std::round instead of std::ceil to improve aspect ratio
+        // in case padding is only slightly larger than h.
+        const unsigned int rounded = std::round(padding / h);
+        // in case padding is much smaller than h, make sure we
+        // have at least 1 element
+        const unsigned int num = (padding > 0. && rounded == 0) ? 1 : rounded;
+        for (unsigned int i = 0; i < num; ++i)
+          step_sizes.push_back(padding / num);
+      };
+
+      std::vector<std::vector<double>> step_sizes(2);
+      // x-coord
+      // left:
+      add_sizes(step_sizes[0], pad_left, outer_radius);
```

Comment:
I was wondering why when you changed from `std::ceil` to `std::round` then the results that you showed went from 4 cells along the right padding direction to 2. I think that here you might need to use 
`add_sizes(step_sizes[0], pad_left, 2*outer_radius)` because `outer_radius` is half the height/width of the block for which the `pad_*` is the full length of. Could you please check this out?

Reply:
> I was wondering why when you changed from std::ceil to std::round then the results that you showed went from 4 cells along the right padding direction to 2.

that's because we want to have cells with as good aspect ratio as possible. So one size (coming from hole-part) is `h`, and padding is `1.2*h`, then `std::ceil` will give 2 cells, whereas the best approach is to have 1 cell.

> I think that here you might need to use
add_sizes(step_sizes[0], pad_left, 2*outer_radius) because outer_radius is half the height/width of the block

nope, `outer_radius` is the dimension of the element coming from hole-mesh which is also half of its height / width.

Reply:
@davydden and I cleared this up: I was mistaken in my assumption that the original mesh created by `hyper_cube_with_cylindrical_hole` has no subdivisions along the length of the bounding box.